frame/dutch-auction/src/lib.rs

```diff
@@ -150,6 +167,30 @@ pub mod pallet {
 			<Self as SellEngine<AuctionStepFunction>>::take(&who, order_id, take)?;
 			Ok(().into())
 		}
+
+		#[pallet::weight(T::WeightInfo::liquidate())]
+		pub fn liquidate(origin: OriginFor<T>, order_id: T::OrderId) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			if let Some(order) = SellOrders::<T>::get(order_id) {
+				if order.from_to == who {
+					// weights fees are of platform spam protection, so we do not interfere with
+					// this function but on pallet level, we allow "fee less" liquidation by owner
+					// we can later allow liquidate old orders(or orders with some block liquidation
+					// timeout set) using kind of account per order is possible, but may risk to
+					// pollute account system
+					let treasury = &T::PalletId::get().into_account();
+					T::NativeCurrency::transfer(
+						&order.from_to,
+						treasury,
```

Comment:
If I understand correctly, we allow someone to cancel an order, in this case, shouldn't we swap the two accounts here because we repay (treasury -> from_to)?

Reply:
oh yes. bug. will fix

Reply:
have we added a unitest to check this bug?

Reply:
i am testing and fixing bug in `dz/weights` branch.

Reply:
and another bug with returning not satisfied takes reservers to accounts

Reply:
So we'll merge this, disable angular in cfg(...), and fix bugs in separate PR?

Reply:
not sure how better. 

disabling angular from runtime if it was enabled will be in separate pr for sure. for now Dutch auction is not called from Lending

Reply:
i will merge branch here.  fixed bugs.